Library/Homebrew/test/formula_auditor_spec.rb

```diff
@@ -714,6 +714,78 @@ class Foo < Formula
       expect(fa.problems).to be_empty
     end
 
+    it "requires `branch:` to be specified for Git head URLs" do
+      fa = formula_auditor "foo", <<~RUBY, online: true
+        class Foo < Formula
+          url "https://brew.sh/foo-1.0.tgz"
+          sha256 "31cccfc6630528db1c8e3a06f6decf2a370060b982841cfab2b8677400a5092e"
+          head "https://github.com/Homebrew/homebrew-test-bot.git"
+        end
+      RUBY
+
+      fa.audit_specs
+      # This is `.last` because the first problem is the unreachable stable URL.
+      expect(fa.problems.last[:message]).to match("Git `head` URL must specify a branch name")
+    end
+
+    it "suggests a detected default branch for Git head URLs" do
+      fa = formula_auditor "foo", <<~RUBY, online: true
+        class Foo < Formula
+          url "https://brew.sh/foo-1.0.tgz"
+          sha256 "31cccfc6630528db1c8e3a06f6decf2a370060b982841cfab2b8677400a5092e"
+          head "https://github.com/Homebrew/homebrew-test-bot.git", branch: "master"
+        end
+      RUBY
+
+      message = "Git `head` URL must specify a branch name - try `branch: \"main\"`"
```

Comment:
I'm a little confused why this is an error, since [homebrew-test-bot has a master branch](https://github.com/Homebrew/homebrew-test-bot/tree/master). I would expect that to be valid.

I think the audit will currently print this error message any time that the specified branch does not match the `HEAD` ref returned by `git ls-remote --symref {url} HEAD`.

Reply:
@issyl0 Thoughts?

Reply:
I looked into this some more and discovered the existence of [audit_exceptions/head_non_default_branch_allowlist.json](https://github.com/Homebrew/homebrew-core/blob/52ec99dae220a432b2c96bf64d2a9f9163046edb/audit_exceptions/head_non_default_branch_allowlist.json). The presence of this allowlist suggests that we _do_ want the HEAD branch to be the default branch, so it should always be suggested, so this audit that I added is working as intended (even though we didn't know this audit would exist at the time we added the allowlist mechanism)? Maybe it just needs a note for core formulae to add to the allowlist?

Reply:
> why not?

@scpeters Because we no longer have a clear majority between `master` and `main`. Most repos are still on `master`. Most new repos use `main`. New versions of `git` default to `main`. Old versions default to `master`. It's a mess.

> Maybe it just needs a note for core formulae to add to the allowlist?

@issyl0 Yes, good idea 👍🏻